### PR TITLE
EIP1-2386 Map Certificate entity to printer's request

### PR DESCRIPTION
**Pull request by Neil-Massey**
Maps the RDS stored certificate and current print request to the Print Request sent to the printer.

### CODE REVIEW

File: src/test/kotlin/uk/gov/dluhc/printapi/rds/mapper/InstantMapperTest.kt [hunk 1]
```diff
@@ -66,4 +67,21 @@ class InstantMapperTest {
...
+            val instant = Instant.now()
+            val expected = instant.atOffset(ZoneOffset.UTC)
+
+            // When
+            val actual = instantMapper.toOffsetDateTime(instant)
+
+            // Then
+            assertThat(actual).isEqualTo(expected)
+        }
+    }
```
Comment (vishalgupta-valtech) [inline src/test/kotlin/uk/gov/dluhc/printapi/rds/mapper/InstantMapperTest.kt:86]: It will be good to also have a test for null instant to null offset date time

File: src/test/kotlin/uk/gov/dluhc/printapi/rds/mapper/CertificateToPrintRequestMapperTest.kt [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+
+class CertificateToPrintRequestMapperTest {
+
+    private lateinit var mapper: CertificateToPrintRequestMapperImpl
+
+    @BeforeEach
+    fun setup() {
+        mapper = CertificateToPrintRequestMapperImpl()
+        ReflectionTestUtils.setField(mapper, "instantMapper", InstantMapper())
+    }
```
Comment (vishalgupta-valtech) [inline src/test/kotlin/uk/gov/dluhc/printapi/rds/mapper/CertificateToPrintRequestMapperTest.kt:42]: This should work

```
@Mock
    private lateinit var instantMapper: InstantMapper

@InjectMocks
    private val mapper = CertificateToPrintRequestMapperImpl()
```

File: src/test/kotlin/uk/gov/dluhc/printapi/rds/mapper/CertificateToPrintRequestMapperTest.kt [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+            gssCode = gssCode,
+            printRequests = mutableListOf(printRequest)
+        )
+        val photoZipPath: String = aPhotoZipPath()
+
+        val expected = uk.gov.dluhc.printapi.printprovider.models.PrintRequest()
+        expected.requestId = requestId
+        expected.issuingAuthorityEn = eroEnglish.name
+        expected.issueDate = LocalDate.parse("2022-10-21")
+        expected.suggestedExpiryDate = LocalDate.parse("2032-10-21")
```
Comment (vishalgupta-valtech) [inline src/test/kotlin/uk/gov/dluhc/printapi/rds/mapper/CertificateToPrintRequestMapperTest.kt:120]: Curious as why can't you set these expected dates to issueDate and suggestedExpiryDate?

File: src/main/kotlin/uk/gov/dluhc/printapi/rds/entity/Certificate.kt [hunk 4]
```diff
@@ -100,19 +98,23 @@ class Certificate(
     }
 
     @PrePersist
-    public fun prePersist() {
+    fun prePersist() {
```
Comment (vishalgupta-valtech) [inline src/main/kotlin/uk/gov/dluhc/printapi/rds/entity/Certificate.kt:101]: fun prePersist could be renamed to something like 'assignStatusBasedOnSomething' (not a blocker)

File: src/test/kotlin/uk/gov/dluhc/printapi/rds/entity/CertificateTest.kt [hunk 5]
```diff
@@ -12,6 +12,49 @@ import java.time.temporal.ChronoUnit
 
 internal class CertificateTest {
 
+    @Nested
+    inner class GetCurrentPrintRequest {
+
+        @Test
+        fun `should get latest print request for Certificate with no Print Requests`() {
+            // Given
+            val certificate = certificateBuilder(printRequests = listOf())
```
Comment (vishalgupta-valtech) [inline src/test/kotlin/uk/gov/dluhc/printapi/rds/entity/CertificateTest.kt:21]: perhaps `emptyList()` ? 😉  not a blocker

File: src/main/kotlin/uk/gov/dluhc/printapi/rds/entity/Certificate.kt (hunk 4 above)
Comment (Neil-Massey) [reply src/main/kotlin/uk/gov/dluhc/printapi/rds/entity/Certificate.kt:101]: Agree - but next PR is removing this method as the status change to the connected entities doesn't trigger the `@PreUpdate` annotation have refactored to set the status in response to request or status being added through the Certificate.

File: src/test/kotlin/uk/gov/dluhc/printapi/rds/entity/CertificateTest.kt (hunk 5 above)
Comment (Neil-Massey) [reply src/test/kotlin/uk/gov/dluhc/printapi/rds/entity/CertificateTest.kt:21]: 👍  - included in next PR

File: src/test/kotlin/uk/gov/dluhc/printapi/rds/mapper/CertificateToPrintRequestMapperTest.kt (hunk 2 above)
Comment (Neil-Massey) [reply src/test/kotlin/uk/gov/dluhc/printapi/rds/mapper/CertificateToPrintRequestMapperTest.kt:42]: Yes - but I chose to use the real object instead of a mock

File: src/test/kotlin/uk/gov/dluhc/printapi/rds/mapper/CertificateToPrintRequestMapperTest.kt (hunk 3 above)
Comment (Neil-Massey) [reply src/test/kotlin/uk/gov/dluhc/printapi/rds/mapper/CertificateToPrintRequestMapperTest.kt:120]: Don't think there is one now - I think elsewhere had issues with Offset v Instant... updated.

File: src/test/kotlin/uk/gov/dluhc/printapi/rds/mapper/InstantMapperTest.kt (hunk 1 above)
Comment (Neil-Massey) [reply src/test/kotlin/uk/gov/dluhc/printapi/rds/mapper/InstantMapperTest.kt:86]: Added
